p2p/libp2p/networksharding/listKadSharder.go

```diff
@@ -147,15 +187,37 @@ func (lks *listKadSharder) IsInterfaceNil() bool {
 	return lks == nil
 }
 
-// computes the kademlia distance between 2 provided peer by doing byte xor operations
-func computeDistance(src peer.ID, dest peer.ID) *big.Int {
-	srcBuff := []byte(src)
-	destBuff := []byte(dest)
+// computes the kademlia distance between 2 provided peer by doing byte xor operations and counting the resulting bits
+func computeDistanceByCountingBits(src peer.ID, dest peer.ID) *big.Int {
+	srcBuff := kbucket.ConvertPeerID(src)
+	destBuff := kbucket.ConvertPeerID(dest)
+
+	cumulatedBits := 0
+	for i := 0; i < len(srcBuff); i++ {
+		result := srcBuff[i] ^ destBuff[i]
+		cumulatedBits += bits.OnesCount8(result)
+	}
+
+	return big.NewInt(0).SetInt64(int64(cumulatedBits))
+}
 
-	result := make([]byte, core.MinInt(len(srcBuff), len(destBuff)))
-	for i := 0; i < len(src) && i < len(dest); i++ {
-		result[i] = srcBuff[i] ^ destBuff[i]
+// computes the kademlia distance between 2 provided peer by doing byte xor operations and applying log2 on the result
+func computeDistanceLog2Based(src peer.ID, dest peer.ID) *big.Int {
+	srcBuff := kbucket.ConvertPeerID(src)
+	destBuff := kbucket.ConvertPeerID(dest)
+
+	val := 0
+	for i := 0; i < len(srcBuff); i++ {
+		result := srcBuff[i] ^ destBuff[i]
+		if result == 0 {
+			val += 8
```

Comment:
No need for if. As leading zero for 0 is 8 in the list. Add always to the wall. Make a break if result != 0

Reply:
Nice catch. :)

Reply:
done